\documentclass[]{book}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\usepackage{hyperref}
\hypersetup{unicode=true,
            pdftitle={16s rRNA analysis workshop},
            pdfauthor={Hena R. Ramay},
            pdfborder={0 0 0},
            breaklinks=true}
\urlstyle{same}  % don't use monospace font for urls
\usepackage{natbib}
\bibliographystyle{apalike}
\usepackage{longtable,booktabs}
\usepackage{graphicx,grffile}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
% Redefines (sub)paragraphs to behave more like sections
\ifx\paragraph\undefined\else
\let\oldparagraph\paragraph
\renewcommand{\paragraph}[1]{\oldparagraph{#1}\mbox{}}
\fi
\ifx\subparagraph\undefined\else
\let\oldsubparagraph\subparagraph
\renewcommand{\subparagraph}[1]{\oldsubparagraph{#1}\mbox{}}
\fi

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\providecommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

  \title{16s rRNA analysis workshop}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
    \author{Hena R. Ramay}
    \preauthor{\centering\large\emph}
  \postauthor{\par}
      \predate{\centering\large\emph}
  \postdate{\par}
    \date{2019-10-06}

\usepackage{booktabs}

\begin{document}
\maketitle

{
\setcounter{tocdepth}{1}
\tableofcontents
}
\hypertarget{introduction}{%
\chapter{Introduction}\label{introduction}}

Our intention is to develop workshop material as we go along. For each day of the workshop, the basic material will be uploaded and more details will be added based on your questions and problems we encounter. So pleae ask as many questions are you can to help us in making this workshop better!!

\hypertarget{workshop-schedule}{%
\section{Workshop Schedule}\label{workshop-schedule}}

We will try to cover the following material in the course:

\begin{itemize}
\tightlist
\item
  Day1

  \begin{itemize}
  \tightlist
  \item
    9:30 - 10:30 am Introductions and a presentation of the basic concepts
  \item
    10:30 - 10:45 am Break
  \item
    10:45 - 12 pm Installations of the required packages and data download
  \item
    12 - 1 pm Lunch break
  \item
    1 - 3 pm Reading data in and inspecting read quality
  \end{itemize}
\item
  Day2

  \begin{itemize}
  \tightlist
  \item
    9:30 -10:30 am Filter and trim + learn error rates + Sample inference + Merge samples
  \item
    10:30 - 10:45 am Break
  \item
    10:45 - 12 Generate sequence table and remove Chimeras
  \item
    12 - 1 pm Lunch break
  \item
    1- 3 pm Taxomnomy explanation + assign your sequences. If time permits, make a phyloseq object
  \end{itemize}
\item
  Day3

  \begin{itemize}
  \tightlist
  \item
    9-10:30 am Using a phyloseq object to calculate alpha and beta diversity
  \item
    10:30 onwards we will use a real world dataset to re-do what we have learned here
  \end{itemize}
\end{itemize}

\hypertarget{important-links}{%
\section{Important links}\label{important-links}}

\begin{itemize}
\tightlist
\item
  DADA2 Tutorial : \href{http://benjjneb.github.io/dada2/tutorial.html}{link}
\item
  Day One presentaton : \href{microbiomeworkshop.pptx}{link}
\item
  Day One dataset : \href{MiSeqSOPData.zip}{link}
\end{itemize}

\hypertarget{day-one}{%
\chapter{Day One}\label{day-one}}

We are very excited to teach this course for the first time and share what we know with you all

So lets start by talking about the very basics:

\hypertarget{what-are-we-trying-to-achieve}{%
\section{What are we trying to achieve}\label{what-are-we-trying-to-achieve}}

Our goal is very similar to gathering data on a city neighbourhood to find out who lives there, how the demographic changes over time or in case of a drastic event. We can gather more information by asking about neighbours, quality of life etc. Similarly when we are looking at microbial communities our first question is who is there, how abundant and how their presence changes over time or when conditions change. We can also ask questions like how the microbiomes are interacting with each other (metabolites).

For the scope of this workshop we will stick to the simple questions: who and how much?

\hypertarget{basics-background}{%
\section{Basics \& Background}\label{basics-background}}

Here is the link to the lecture we will start with today: \href{microbiomeworkshop.pptx}{workshop}

Key points are:

\begin{itemize}
\tightlist
\item
  Think of a hypothesis before doing an experiment
\item
  Spend time on experiment design.

  \begin{itemize}
  \tightlist
  \item
    Sample size, 16s region to amplify etc
  \item
    Talk to a bioinformatician
  \item
    Think about the depth of sequencing if you want to capture the less abundant taxa
  \item
    Add negative control to account for contamination
  \end{itemize}
\item
  Thoughtful data analysis is critical for successful identification of microbes
\end{itemize}

``If you torture the data long enough, it will confess.''- Ronald Coase, Economist

\hypertarget{dada2-pipeline-v1.2}{%
\section{DADA2 pipeline (v1.2)}\label{dada2-pipeline-v1.2}}

From now on, we will be working on the DADA2 package version 1.12. DADA2 has great documentation and an excellent tutorial online that we will use to understand the pipeline. Please go to the following link \url{http://benjjneb.github.io/dada2/tutorial.html}

\hypertarget{data-for-the-tutorial}{%
\subsection{Data for the tutorial}\label{data-for-the-tutorial}}

The data to use for the tutorial can be downloaded from \href{MiSeqSOPData.zip}{here}

\hypertarget{getting-ready-load-packages-and-get-file-list}{%
\subsection{Getting ready ( load packages and get file list)}\label{getting-ready-load-packages-and-get-file-list}}

functions that we will be using here are :

\begin{itemize}
\tightlist
\item
  list.files()
\item
  sort()
\item
  strsplit()
\item
  basename()
\item
  sapply()
\end{itemize}

\hypertarget{inspect-read-quality-profiles}{%
\section{Inspect read quality profiles}\label{inspect-read-quality-profiles}}

\hypertarget{day-two}{%
\chapter{Day Two}\label{day-two}}

We describe our methods in this chapter.

\hypertarget{day-three}{%
\chapter{Day Three}\label{day-three}}

\hypertarget{summary}{%
\chapter{Summary}\label{summary}}

\bibliography{book.bib,packages.bib}


\end{document}
