<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 Day Two | 16s rRNA analysis workshop</title>
  <meta name="description" content="This tutorial covers workshop material for all three days" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="5 Day Two | 16s rRNA analysis workshop" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This tutorial covers workshop material for all three days" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 Day Two | 16s rRNA analysis workshop" />
  
  <meta name="twitter:description" content="This tutorial covers workshop material for all three days" />
  

<meta name="author" content="Hena R. Ramay" />


<meta name="date" content="2022-09-30" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="dada2.html"/>
<link rel="next" href="project.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">16s rRNA Analysis Workshop</a></li>

<li class="divider"></li>
<li><a href="index.html#introduction" id="toc-introduction"><span class="toc-section-number">1</span> Introduction</a>
<ul>
<li><a href="index.html#workshop-schedule" id="toc-workshop-schedule"><span class="toc-section-number">1.1</span> Workshop Schedule</a></li>
<li><a href="index.html#important-links" id="toc-important-links"><span class="toc-section-number">1.2</span> Important links</a></li>
</ul></li>
<li><a href="basic-information.html#basic-information" id="toc-basic-information"><span class="toc-section-number">2</span> Basic Information</a>
<ul>
<li><a href="basic-information.html#what-are-we-trying-to-achieve" id="toc-what-are-we-trying-to-achieve"><span class="toc-section-number">2.1</span> What are we trying to achieve</a></li>
<li><a href="basic-information.html#basics-background" id="toc-basics-background"><span class="toc-section-number">2.2</span> Basics &amp; Background</a></li>
</ul></li>
<li><a href="primer-removal.html#primer-removal" id="toc-primer-removal"><span class="toc-section-number">3</span> Primer Removal</a>
<ul>
<li><a href="primer-removal.html#cutadapt" id="toc-cutadapt"><span class="toc-section-number">3.1</span> CutAdapt</a>
<ul>
<li><a href="primer-removal.html#primer-removal-for-paired-end-reads" id="toc-primer-removal-for-paired-end-reads"><span class="toc-section-number">3.1.1</span> Primer removal for paired end reads</a></li>
<li><a href="primer-removal.html#quality-trimming" id="toc-quality-trimming"><span class="toc-section-number">3.1.2</span> Quality trimming</a></li>
<li><a href="primer-removal.html#call-cutadapt-from-r" id="toc-call-cutadapt-from-r"><span class="toc-section-number">3.1.3</span> Call cutAdapt from R</a></li>
</ul></li>
</ul></li>
<li><a href="dada2.html#dada2" id="toc-dada2"><span class="toc-section-number">4</span> DADA2</a>
<ul>
<li><a href="dada2.html#dada2-pipeline-v1.2" id="toc-dada2-pipeline-v1.2"><span class="toc-section-number">4.1</span> DADA2 pipeline (v1.2)</a>
<ul>
<li><a href="dada2.html#data-for-the-tutorial" id="toc-data-for-the-tutorial"><span class="toc-section-number">4.1.1</span> Data for the tutorial</a></li>
<li><a href="dada2.html#getting-ready-load-packages-and-get-file-list" id="toc-getting-ready-load-packages-and-get-file-list"><span class="toc-section-number">4.1.2</span> Getting ready ( load packages and get file list)</a></li>
<li><a href="dada2.html#inspect-read-quality-profiles" id="toc-inspect-read-quality-profiles"><span class="toc-section-number">4.1.3</span> Inspect read quality profiles</a></li>
<li><a href="dada2.html#lets-make-plots-to-look-at-the-quality-for-multiple-files" id="toc-lets-make-plots-to-look-at-the-quality-for-multiple-files"><span class="toc-section-number">4.1.4</span> Lets make plots to look at the quality for multiple files</a></li>
</ul></li>
<li><a href="dada2.html#some-tips" id="toc-some-tips"><span class="toc-section-number">4.2</span> Some tips:</a></li>
</ul></li>
<li><a href="day-two.html#day-two" id="toc-day-two"><span class="toc-section-number">5</span> Day Two</a>
<ul>
<li><a href="day-two.html#main-concepts-to-be-discussed" id="toc-main-concepts-to-be-discussed"><span class="toc-section-number">5.1</span> Main concepts to be discussed:</a></li>
<li><a href="day-two.html#alpha-beta-diversity" id="toc-alpha-beta-diversity"><span class="toc-section-number">5.2</span> Alpha &amp; beta Diversity</a></li>
</ul></li>
<li><a href="project.html#project" id="toc-project"><span class="toc-section-number">6</span> Project</a>
<ul>
<li><a href="project.html#setup" id="toc-setup"><span class="toc-section-number">6.1</span> Setup</a></li>
<li><a href="project.html#task-1-ignore-this-task" id="toc-task-1-ignore-this-task"><span class="toc-section-number">6.2</span> Task 1 (Ignore this task)</a>
<ul>
<li><a href="project.html#cutadapt-1" id="toc-cutadapt-1"><span class="toc-section-number">6.2.1</span> Cutadapt</a></li>
<li><a href="project.html#trimmomatic" id="toc-trimmomatic"><span class="toc-section-number">6.2.2</span> Trimmomatic</a></li>
<li><a href="project.html#primers-removed" id="toc-primers-removed"><span class="toc-section-number">6.2.3</span> Primers removed</a></li>
</ul></li>
<li><a href="project.html#task-2" id="toc-task-2"><span class="toc-section-number">6.3</span> Task 2</a></li>
<li><a href="project.html#task-3" id="toc-task-3"><span class="toc-section-number">6.4</span> Task 3</a></li>
<li><a href="project.html#project-solutions" id="toc-project-solutions"><span class="toc-section-number">6.5</span> Project solutions</a>
<ul>
<li><a href="project.html#task-2---dada2" id="toc-task-2---dada2"><span class="toc-section-number">6.5.1</span> Task 2 - dada2</a></li>
<li><a href="project.html#task-3---phyloseq" id="toc-task-3---phyloseq"><span class="toc-section-number">6.5.2</span> Task 3 - phyloseq</a></li>
</ul></li>
</ul></li>
<li><a href="what-next.html#what-next" id="toc-what-next"><span class="toc-section-number">7</span> What Next?</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">16s rRNA analysis workshop</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="day-two" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Day Two</h1>
<div id="main-concepts-to-be-discussed" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Main concepts to be discussed:</h2>
<ul>
<li>Finish dada2 pipeline</li>
<li>Assign Taxonomy</li>
<li>Intro to Phyolseq package</li>
<li>Create a Phylum level bar plots</li>
<li>Alpha diversity plots</li>
<li>Beta diversity plots</li>
</ul>
<p>Once we are finished using the dada2 package, we will have a sequence table and taxonomy table.</p>
<p>Lets look at the metadata files we have to get more information about these samples.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="day-two.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb10-2"><a href="day-two.html#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="day-two.html#cb14-1" aria-hidden="true" tabindex="-1"></a>time_info<span class="ot">&lt;-</span><span class="fu">read.delim</span>(<span class="st">&quot;~/projects/16sanalysis_workshop/workshoptutorial/MiSeq_SOP/mouse.time.design&quot;</span>,<span class="at">sep  =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb14-2"><a href="day-two.html#cb14-2" aria-hidden="true" tabindex="-1"></a>dpw_info<span class="ot">&lt;-</span><span class="fu">read.delim</span>(<span class="st">&quot;~/projects/16sanalysis_workshop/workshoptutorial/MiSeq_SOP/mouse.dpw.metadata&quot;</span>,<span class="at">sep  =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb14-3"><a href="day-two.html#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="day-two.html#cb14-4" aria-hidden="true" tabindex="-1"></a>sample_info<span class="ot">&lt;-</span><span class="fu">left_join</span>(time_info,dpw_info,<span class="at">by=</span><span class="st">&quot;group&quot;</span>)</span>
<span id="cb14-5"><a href="day-two.html#cb14-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-6"><a href="day-two.html#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sample_info)<span class="ot">&lt;-</span> sample_info<span class="sc">$</span>group</span>
<span id="cb14-7"><a href="day-two.html#cb14-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-8"><a href="day-two.html#cb14-8" aria-hidden="true" tabindex="-1"></a>sample_info</span></code></pre></div>
<pre><code>##         group  time dpw
## F3D0     F3D0 Early   0
## F3D1     F3D1 Early   1
## F3D141 F3D141  Late 141
## F3D142 F3D142  Late 142
## F3D143 F3D143  Late 143
## F3D144 F3D144  Late 144
## F3D145 F3D145  Late 145
## F3D146 F3D146  Late 146
## F3D147 F3D147  Late 147
## F3D148 F3D148  Late 148
## F3D149 F3D149  Late 149
## F3D150 F3D150  Late 150
## F3D2     F3D2 Early   2
## F3D3     F3D3 Early   3
## F3D5     F3D5 Early   5
## F3D6     F3D6 Early   6
## F3D7     F3D7 Early   7
## F3D8     F3D8 Early   8
## F3D9     F3D9 Early   9</code></pre>
<p>Lets make a phyloseq object</p>
<p>Now that we have sample_info lets try to make a phyloseq object out of this</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="day-two.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb16-2"><a href="day-two.html#cb16-2" aria-hidden="true" tabindex="-1"></a>taxa<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="st">&quot;~/projects/16sanalysis_workshop/workshoptutorial/output/taxa.rds&quot;</span>)</span>
<span id="cb16-3"><a href="day-two.html#cb16-3" aria-hidden="true" tabindex="-1"></a>seq<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="st">&quot;~/projects/16sanalysis_workshop/workshoptutorial/output/seq.rds&quot;</span>)</span>
<span id="cb16-4"><a href="day-two.html#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="day-two.html#cb16-5" aria-hidden="true" tabindex="-1"></a>ps <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(<span class="fu">otu_table</span>(seq,<span class="at">taxa_are_rows =</span> F),</span>
<span id="cb16-6"><a href="day-two.html#cb16-6" aria-hidden="true" tabindex="-1"></a>               <span class="fu">sample_data</span>(sample_info),</span>
<span id="cb16-7"><a href="day-two.html#cb16-7" aria-hidden="true" tabindex="-1"></a>               <span class="fu">tax_table</span>(taxa))</span>
<span id="cb16-8"><a href="day-two.html#cb16-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-9"><a href="day-two.html#cb16-9" aria-hidden="true" tabindex="-1"></a>ps</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 234 taxa and 19 samples ]
## sample_data() Sample Data:       [ 19 samples by 3 sample variables ]
## tax_table()   Taxonomy Table:    [ 234 taxa by 7 taxonomic ranks ]</code></pre>
<p>You can see that this object has an OTU table(ASV table), sample data and tax_table. You can use functions tax_table(), sample_data() and otu_table() to access the data.</p>
<p>Take a look at:</p>
<ul>
<li>subset_samples()</li>
<li>subset_taxa()</li>
<li>tax_glom()</li>
<li>sample_sums()</li>
<li>prune_samples()</li>
<li>transform_sample_counts()</li>
<li>psmelt()</li>
</ul>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="day-two.html#cb18-1" aria-hidden="true" tabindex="-1"></a>ps2<span class="ot">&lt;-</span><span class="fu">tax_glom</span>(ps,<span class="at">taxrank =</span> <span class="st">&quot;Phylum&quot;</span>)</span>
<span id="cb18-2"><a href="day-two.html#cb18-2" aria-hidden="true" tabindex="-1"></a>ps2 <span class="ot">=</span> <span class="fu">transform_sample_counts</span>(ps2, <span class="cf">function</span>(x) x<span class="sc">/</span><span class="fu">sum</span>(x))</span>
<span id="cb18-3"><a href="day-two.html#cb18-3" aria-hidden="true" tabindex="-1"></a>pmelt<span class="ot">&lt;-</span><span class="fu">psmelt</span>(ps2) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(Abundance))</span>
<span id="cb18-4"><a href="day-two.html#cb18-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-5"><a href="day-two.html#cb18-5" aria-hidden="true" tabindex="-1"></a>cutoff<span class="ot">&lt;-</span><span class="fl">0.005</span></span>
<span id="cb18-6"><a href="day-two.html#cb18-6" aria-hidden="true" tabindex="-1"></a>pmelt_filt<span class="ot">&lt;-</span>pmelt <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Phylum,time) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">sum</span>(Abundance) <span class="sc">&gt;</span>cutoff)</span>
<span id="cb18-7"><a href="day-two.html#cb18-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-8"><a href="day-two.html#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(pmelt_filt,<span class="fu">aes</span>(<span class="at">x=</span>Sample,<span class="at">y=</span>Abundance,<span class="at">color=</span>Phylum,<span class="at">fill=</span>Phylum)) <span class="sc">+</span><span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>)</span></code></pre></div>
<p><img src="16sworkshop_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="day-two.html#cb19-1" aria-hidden="true" tabindex="-1"></a>pmelt_filt <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(time,Phylum) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">mean_Abundance=</span><span class="fu">mean</span>(Abundance)) <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(.,<span class="fu">aes</span>(<span class="at">x=</span>time,<span class="at">y=</span>mean_Abundance,<span class="at">fill=</span>Phylum)) <span class="sc">+</span><span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>)</span></code></pre></div>
<pre><code>## `summarise()` has grouped output by &#39;time&#39;. You can override using the
## `.groups` argument.</code></pre>
<p><img src="16sworkshop_files/figure-html/unnamed-chunk-2-2.png" width="672" /></p>
<p>Figure out how to change the colors to custom colors!!</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="day-two.html#cb21-1" aria-hidden="true" tabindex="-1"></a>pmelt_filt <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(time,Phylum) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(Abundance))</span></code></pre></div>
<pre><code>## `summarise()` has grouped output by &#39;time&#39;. You can override using the
## `.groups` argument.</code></pre>
<pre><code>## # A tibble: 12 Ã— 3
## # Groups:   time [2]
##    time  Phylum              mean
##    &lt;chr&gt; &lt;chr&gt;              &lt;dbl&gt;
##  1 Early Actinobacteria  0.00133 
##  2 Early Bacteroidetes   0.592   
##  3 Early Cyanobacteria   0.000594
##  4 Early Firmicutes      0.378   
##  5 Early Patescibacteria 0.00285 
##  6 Early Proteobacteria  0.00331 
##  7 Early Tenericutes     0.0218  
##  8 Late  Actinobacteria  0.00472 
##  9 Late  Bacteroidetes   0.716   
## 10 Late  Firmicutes      0.269   
## 11 Late  Patescibacteria 0.000993
## 12 Late  Tenericutes     0.00845</code></pre>
</div>
<div id="alpha-beta-diversity" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Alpha &amp; beta Diversity</h2>
<p><img src="16sworkshop_files/figure-html/alpha%20diversit-1.png" width="672" /></p>
<p><img src="16sworkshop_files/figure-html/unnamed-chunk-3-1.png" width="672" /><img src="16sworkshop_files/figure-html/unnamed-chunk-3-2.png" width="672" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="dada2.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="project.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["16sworkshop.pdf", "16sworkshop.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
