<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 Downstream Analysis | 16s rRNA analysis workshop</title>
  <meta name="description" content="This tutorial covers workshop material for all three days" />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="5 Downstream Analysis | 16s rRNA analysis workshop" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This tutorial covers workshop material for all three days" />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 Downstream Analysis | 16s rRNA analysis workshop" />
  
  <meta name="twitter:description" content="This tutorial covers workshop material for all three days" />
  

<meta name="author" content="Hena R. Ramay" />


<meta name="date" content="2022-10-02" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="dada2.html"/>
<link rel="next" href="project.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">16s rRNA Analysis Workshop</a></li>

<li class="divider"></li>
<li><a href="index.html#introduction" id="toc-introduction"><span class="toc-section-number">1</span> Introduction</a>
<ul>
<li><a href="index.html#workshop-schedule" id="toc-workshop-schedule"><span class="toc-section-number">1.1</span> Workshop Schedule</a></li>
<li><a href="index.html#important-links" id="toc-important-links"><span class="toc-section-number">1.2</span> Important links</a></li>
</ul></li>
<li><a href="basic-information.html#basic-information" id="toc-basic-information"><span class="toc-section-number">2</span> Basic Information</a>
<ul>
<li><a href="basic-information.html#what-are-we-trying-to-achieve" id="toc-what-are-we-trying-to-achieve"><span class="toc-section-number">2.1</span> What are we trying to achieve</a></li>
<li><a href="basic-information.html#basics-background" id="toc-basics-background"><span class="toc-section-number">2.2</span> Basics &amp; Background</a></li>
</ul></li>
<li><a href="primer-removal.html#primer-removal" id="toc-primer-removal"><span class="toc-section-number">3</span> Primer Removal</a>
<ul>
<li><a href="primer-removal.html#cutadapt" id="toc-cutadapt"><span class="toc-section-number">3.1</span> CutAdapt</a>
<ul>
<li><a href="primer-removal.html#primer-removal-for-paired-end-reads" id="toc-primer-removal-for-paired-end-reads"><span class="toc-section-number">3.1.1</span> Primer removal for paired end reads</a></li>
<li><a href="primer-removal.html#quality-trimming" id="toc-quality-trimming"><span class="toc-section-number">3.1.2</span> Quality trimming</a></li>
<li><a href="primer-removal.html#call-cutadapt-from-r" id="toc-call-cutadapt-from-r"><span class="toc-section-number">3.1.3</span> Call cutAdapt from R</a></li>
</ul></li>
</ul></li>
<li><a href="dada2.html#dada2" id="toc-dada2"><span class="toc-section-number">4</span> DADA2</a>
<ul>
<li><a href="dada2.html#dada2-pipeline-v1.18" id="toc-dada2-pipeline-v1.18"><span class="toc-section-number">4.1</span> DADA2 pipeline (v1.18)</a>
<ul>
<li><a href="dada2.html#data-for-the-tutorial" id="toc-data-for-the-tutorial"><span class="toc-section-number">4.1.1</span> Data for the tutorial</a></li>
<li><a href="dada2.html#getting-ready-load-packages-and-get-file-list" id="toc-getting-ready-load-packages-and-get-file-list"><span class="toc-section-number">4.1.2</span> Getting ready ( load packages and get file list)</a></li>
<li><a href="dada2.html#inspect-read-quality-profiles" id="toc-inspect-read-quality-profiles"><span class="toc-section-number">4.1.3</span> Inspect read quality profiles</a></li>
<li><a href="dada2.html#lets-make-plots-to-look-at-the-quality-for-multiple-files" id="toc-lets-make-plots-to-look-at-the-quality-for-multiple-files"><span class="toc-section-number">4.1.4</span> Lets make plots to look at the quality for multiple files</a></li>
</ul></li>
<li><a href="dada2.html#some-tips" id="toc-some-tips"><span class="toc-section-number">4.2</span> Some tips:</a></li>
</ul></li>
<li><a href="downstream-analysis.html#downstream-analysis" id="toc-downstream-analysis"><span class="toc-section-number">5</span> Downstream Analysis</a>
<ul>
<li><a href="downstream-analysis.html#phyloseq" id="toc-phyloseq"><span class="toc-section-number">5.1</span> Phyloseq</a></li>
<li><a href="downstream-analysis.html#alpha-diversity" id="toc-alpha-diversity"><span class="toc-section-number">5.2</span> Alpha Diversity</a></li>
<li><a href="downstream-analysis.html#beta-diversity" id="toc-beta-diversity"><span class="toc-section-number">5.3</span> Beta Diversity</a></li>
<li><a href="downstream-analysis.html#differential-abundance-analysis" id="toc-differential-abundance-analysis"><span class="toc-section-number">5.4</span> Differential Abundance Analysis</a>
<ul>
<li><a href="downstream-analysis.html#deseq2" id="toc-deseq2"><span class="toc-section-number">5.4.1</span> DESEQ2</a></li>
</ul></li>
</ul></li>
<li><a href="project.html#project" id="toc-project"><span class="toc-section-number">6</span> Project</a>
<ul>
<li><a href="project.html#setup" id="toc-setup"><span class="toc-section-number">6.1</span> Setup</a></li>
<li><a href="project.html#task-1-ignore-this-task" id="toc-task-1-ignore-this-task"><span class="toc-section-number">6.2</span> Task 1 (Ignore this task)</a>
<ul>
<li><a href="project.html#cutadapt-1" id="toc-cutadapt-1"><span class="toc-section-number">6.2.1</span> Cutadapt</a></li>
<li><a href="project.html#trimmomatic" id="toc-trimmomatic"><span class="toc-section-number">6.2.2</span> Trimmomatic</a></li>
<li><a href="project.html#primers-removed" id="toc-primers-removed"><span class="toc-section-number">6.2.3</span> Primers removed</a></li>
</ul></li>
<li><a href="project.html#task-2" id="toc-task-2"><span class="toc-section-number">6.3</span> Task 2</a></li>
<li><a href="project.html#task-3" id="toc-task-3"><span class="toc-section-number">6.4</span> Task 3</a></li>
<li><a href="project.html#project-solutions" id="toc-project-solutions"><span class="toc-section-number">6.5</span> Project solutions</a>
<ul>
<li><a href="project.html#task-2---dada2" id="toc-task-2---dada2"><span class="toc-section-number">6.5.1</span> Task 2 - dada2</a></li>
<li><a href="project.html#task-3---phyloseq" id="toc-task-3---phyloseq"><span class="toc-section-number">6.5.2</span> Task 3 - phyloseq</a></li>
</ul></li>
</ul></li>
<li><a href="what-next.html#what-next" id="toc-what-next"><span class="toc-section-number">7</span> What Next?</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">16s rRNA analysis workshop</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="downstream-analysis" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Downstream Analysis</h1>
<div id="phyloseq" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Phyloseq</h2>
<p>Once we are finished using the dada2 package, we will have a sequence table and taxonomy table.</p>
<p>Lets look at the metadata files we have to get more information about these samples.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="downstream-analysis.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb10-2"><a href="downstream-analysis.html#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="downstream-analysis.html#cb14-1" aria-hidden="true" tabindex="-1"></a>time_info<span class="ot">&lt;-</span><span class="fu">read.delim</span>(<span class="st">&quot;~/projects/16sanalysis_workshop/workshoptutorial/MiSeq_SOP/mouse.time.design&quot;</span>,<span class="at">sep  =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb14-2"><a href="downstream-analysis.html#cb14-2" aria-hidden="true" tabindex="-1"></a>dpw_info<span class="ot">&lt;-</span><span class="fu">read.delim</span>(<span class="st">&quot;~/projects/16sanalysis_workshop/workshoptutorial/MiSeq_SOP/mouse.dpw.metadata&quot;</span>,<span class="at">sep  =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb14-3"><a href="downstream-analysis.html#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="downstream-analysis.html#cb14-4" aria-hidden="true" tabindex="-1"></a>sample_info<span class="ot">&lt;-</span><span class="fu">left_join</span>(time_info,dpw_info,<span class="at">by=</span><span class="st">&quot;group&quot;</span>)</span>
<span id="cb14-5"><a href="downstream-analysis.html#cb14-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-6"><a href="downstream-analysis.html#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sample_info)<span class="ot">&lt;-</span> sample_info<span class="sc">$</span>group</span>
<span id="cb14-7"><a href="downstream-analysis.html#cb14-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-8"><a href="downstream-analysis.html#cb14-8" aria-hidden="true" tabindex="-1"></a>sample_info</span></code></pre></div>
<pre><code>##         group  time dpw
## F3D0     F3D0 Early   0
## F3D1     F3D1 Early   1
## F3D141 F3D141  Late 141
## F3D142 F3D142  Late 142
## F3D143 F3D143  Late 143
## F3D144 F3D144  Late 144
## F3D145 F3D145  Late 145
## F3D146 F3D146  Late 146
## F3D147 F3D147  Late 147
## F3D148 F3D148  Late 148
## F3D149 F3D149  Late 149
## F3D150 F3D150  Late 150
## F3D2     F3D2 Early   2
## F3D3     F3D3 Early   3
## F3D5     F3D5 Early   5
## F3D6     F3D6 Early   6
## F3D7     F3D7 Early   7
## F3D8     F3D8 Early   8
## F3D9     F3D9 Early   9</code></pre>
<p>Lets make a phyloseq object</p>
<p>Now that we have sample_info lets try to make a phyloseq object out of this</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="downstream-analysis.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb16-2"><a href="downstream-analysis.html#cb16-2" aria-hidden="true" tabindex="-1"></a>taxa<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="st">&quot;~/projects/16sanalysis_workshop/workshoptutorial/output/taxa.rds&quot;</span>)</span>
<span id="cb16-3"><a href="downstream-analysis.html#cb16-3" aria-hidden="true" tabindex="-1"></a>seq<span class="ot">&lt;-</span><span class="fu">readRDS</span>(<span class="st">&quot;~/projects/16sanalysis_workshop/workshoptutorial/output/seq.rds&quot;</span>)</span>
<span id="cb16-4"><a href="downstream-analysis.html#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="downstream-analysis.html#cb16-5" aria-hidden="true" tabindex="-1"></a>ps <span class="ot">&lt;-</span> <span class="fu">phyloseq</span>(<span class="fu">otu_table</span>(seq,<span class="at">taxa_are_rows =</span> F),</span>
<span id="cb16-6"><a href="downstream-analysis.html#cb16-6" aria-hidden="true" tabindex="-1"></a>               <span class="fu">sample_data</span>(sample_info),</span>
<span id="cb16-7"><a href="downstream-analysis.html#cb16-7" aria-hidden="true" tabindex="-1"></a>               <span class="fu">tax_table</span>(taxa))</span>
<span id="cb16-8"><a href="downstream-analysis.html#cb16-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-9"><a href="downstream-analysis.html#cb16-9" aria-hidden="true" tabindex="-1"></a>ps</span></code></pre></div>
<pre><code>## phyloseq-class experiment-level object
## otu_table()   OTU Table:         [ 234 taxa and 19 samples ]
## sample_data() Sample Data:       [ 19 samples by 3 sample variables ]
## tax_table()   Taxonomy Table:    [ 234 taxa by 7 taxonomic ranks ]</code></pre>
<p>You can see that this object has an OTU table(ASV table), sample data and tax_table. You can use functions tax_table(), sample_data() and otu_table() to access the data.</p>
<p>Take a look at:</p>
<ul>
<li>subset_samples()</li>
<li>subset_taxa()</li>
<li>tax_glom()</li>
<li>sample_sums()</li>
<li>prune_samples()</li>
<li>transform_sample_counts()</li>
<li>psmelt()</li>
</ul>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="downstream-analysis.html#cb18-1" aria-hidden="true" tabindex="-1"></a>ps2<span class="ot">&lt;-</span><span class="fu">tax_glom</span>(ps,<span class="at">taxrank =</span> <span class="st">&quot;Phylum&quot;</span>)</span>
<span id="cb18-2"><a href="downstream-analysis.html#cb18-2" aria-hidden="true" tabindex="-1"></a>ps2 <span class="ot">=</span> <span class="fu">transform_sample_counts</span>(ps2, <span class="cf">function</span>(x) x<span class="sc">/</span><span class="fu">sum</span>(x))</span>
<span id="cb18-3"><a href="downstream-analysis.html#cb18-3" aria-hidden="true" tabindex="-1"></a>pmelt<span class="ot">&lt;-</span><span class="fu">psmelt</span>(ps2) <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(Abundance))</span>
<span id="cb18-4"><a href="downstream-analysis.html#cb18-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-5"><a href="downstream-analysis.html#cb18-5" aria-hidden="true" tabindex="-1"></a>cutoff<span class="ot">&lt;-</span><span class="fl">0.005</span></span>
<span id="cb18-6"><a href="downstream-analysis.html#cb18-6" aria-hidden="true" tabindex="-1"></a>pmelt_filt<span class="ot">&lt;-</span>pmelt <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Phylum,time) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">sum</span>(Abundance) <span class="sc">&gt;</span>cutoff)</span>
<span id="cb18-7"><a href="downstream-analysis.html#cb18-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-8"><a href="downstream-analysis.html#cb18-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(pmelt_filt,<span class="fu">aes</span>(<span class="at">x=</span>Sample,<span class="at">y=</span>Abundance,<span class="at">color=</span>Phylum,<span class="at">fill=</span>Phylum)) <span class="sc">+</span><span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>)</span></code></pre></div>
<p><img src="16sworkshop_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="downstream-analysis.html#cb19-1" aria-hidden="true" tabindex="-1"></a>pmelt_filt <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(time,Phylum) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">mean_Abundance=</span><span class="fu">mean</span>(Abundance)) <span class="sc">%&gt;%</span> <span class="fu">ggplot</span>(.,<span class="fu">aes</span>(<span class="at">x=</span>time,<span class="at">y=</span>mean_Abundance,<span class="at">fill=</span>Phylum)) <span class="sc">+</span><span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">&quot;identity&quot;</span>)</span></code></pre></div>
<pre><code>## `summarise()` has grouped output by &#39;time&#39;. You can override using the
## `.groups` argument.</code></pre>
<p><img src="16sworkshop_files/figure-html/unnamed-chunk-2-2.png" width="672" /></p>
<p>Figure out how to change the colors to custom colors!!</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="downstream-analysis.html#cb21-1" aria-hidden="true" tabindex="-1"></a>pmelt_filt <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(time,Phylum) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">mean=</span><span class="fu">mean</span>(Abundance))</span></code></pre></div>
<pre><code>## `summarise()` has grouped output by &#39;time&#39;. You can override using the
## `.groups` argument.</code></pre>
<pre><code>## # A tibble: 12 × 3
## # Groups:   time [2]
##    time  Phylum              mean
##    &lt;chr&gt; &lt;chr&gt;              &lt;dbl&gt;
##  1 Early Actinobacteria  0.00133 
##  2 Early Bacteroidetes   0.592   
##  3 Early Cyanobacteria   0.000594
##  4 Early Firmicutes      0.378   
##  5 Early Patescibacteria 0.00285 
##  6 Early Proteobacteria  0.00331 
##  7 Early Tenericutes     0.0218  
##  8 Late  Actinobacteria  0.00472 
##  9 Late  Bacteroidetes   0.716   
## 10 Late  Firmicutes      0.269   
## 11 Late  Patescibacteria 0.000993
## 12 Late  Tenericutes     0.00845</code></pre>
</div>
<div id="alpha-diversity" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> Alpha Diversity</h2>
<p><strong>Alpha Diversity:</strong> Variety and abundance of organisms within a sample</p>
<p><strong>Species richness:</strong> The number of different kinds of organisms present in a particular community</p>
<p><strong>Species evenness</strong> Compares the uniformity of the population size of each of the species present</p>
<p>There are different alpha diversity measures to calculate the biodiversity including Simpson, Shannon, and Observed. These indexes show different estimates of magnitude of change in the niche giving us a more complete picture. There are differences in these indexes in weighting and unit:</p>
<p><strong>Simpson:</strong> Estimator of species richness and species evenness, but provides more weight to evenness resulted from the sum of squared proportions.</p>
<p><strong>Shannon:</strong> Estimator of species richness and species evenness, but provides more weight to richness than evenness and is shown as a logarithmic value.</p>
<p><strong>Observed:</strong> Abundance-based estimator of species richness. This index calculates the minimal number of OTUs (rare OTUs) present in a sample. This index gives more weight to the low abundant species.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="downstream-analysis.html#cb24-1" aria-hidden="true" tabindex="-1"></a>richness<span class="ot">&lt;-</span><span class="fu">estimate_richness</span>(ps)</span>
<span id="cb24-2"><a href="downstream-analysis.html#cb24-2" aria-hidden="true" tabindex="-1"></a>richness<span class="sc">$</span>time<span class="ot">&lt;-</span>ps<span class="sc">@</span>sam_data<span class="sc">$</span>time</span>
<span id="cb24-3"><a href="downstream-analysis.html#cb24-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-4"><a href="downstream-analysis.html#cb24-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(richness,<span class="fu">aes</span>(<span class="at">x=</span>time,<span class="at">y=</span>Shannon,<span class="at">color=</span>time)) <span class="sc">+</span> <span class="fu">geom_boxplot</span>() <span class="sc">+</span><span class="fu">geom_jitter</span>()</span></code></pre></div>
<p><img src="16sworkshop_files/figure-html/alpha%20diversit-1.png" width="672" /></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="downstream-analysis.html#cb25-1" aria-hidden="true" tabindex="-1"></a>wilcoxon_shannon <span class="ot">&lt;-</span> <span class="fu">wilcox.test</span>(Shannon <span class="sc">~</span> time, <span class="at">data =</span> richness)</span>
<span id="cb25-2"><a href="downstream-analysis.html#cb25-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-3"><a href="downstream-analysis.html#cb25-3" aria-hidden="true" tabindex="-1"></a>wilcoxon_shannon</span></code></pre></div>
<pre><code>## 
##  Wilcoxon rank sum exact test
## 
## data:  Shannon by time
## W = 65, p-value = 0.1128
## alternative hypothesis: true location shift is not equal to 0</code></pre>
</div>
<div id="beta-diversity" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> Beta Diversity</h2>
<p>Beta diversity is a measure of dissimilarity in microbial composition <strong>between</strong> two samples.</p>
<p>Distance metrics: Bray–Curtis method does not look at the phylogenetic relatedness and takes the taxa abundance into consideration. This distance is defined as the difference of the abundance divided by the total abundance contributed by both samples. Unifrac method calculates the distances between samples based on their phylogenetic relatedness. There are two unifrac measures: unweighted UniFrac, a qualitative measure that takes presence/absence of data into consideration to compare community composition. This suggests if ecological factors prevent a taxonomic group from occupying certain habitats.Weighted UniFrac, a quantitative measure weights the branches of a phylogenetic tree based on the the the relative abundance of species/taxa. This suggests if ecological differences between habitats have caused the abundance of taxonomic groups to change.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="downstream-analysis.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate Bray-curtis dissimilarity</span></span>
<span id="cb27-2"><a href="downstream-analysis.html#cb27-2" aria-hidden="true" tabindex="-1"></a>ord<span class="ot">&lt;-</span><span class="fu">ordinate</span>(ps,<span class="at">method =</span> <span class="st">&quot;PCoA&quot;</span>,<span class="at">distance =</span> <span class="st">&quot;bray&quot;</span>)</span>
<span id="cb27-3"><a href="downstream-analysis.html#cb27-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-4"><a href="downstream-analysis.html#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="co"># plot ordination</span></span>
<span id="cb27-5"><a href="downstream-analysis.html#cb27-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_ordination</span>(ps,<span class="at">ordination =</span> ord,<span class="at">color=</span><span class="st">&quot;time&quot;</span>,<span class="at">title=</span><span class="st">&quot;Bray PCoA (No transformation)&quot;</span>)<span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values=</span>(<span class="fu">c</span>(<span class="st">&quot;tomato&quot;</span>,<span class="st">&quot;blue&quot;</span>)))</span></code></pre></div>
<p><img src="16sworkshop_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="downstream-analysis.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot ordination with relative abundance values</span></span>
<span id="cb28-2"><a href="downstream-analysis.html#cb28-2" aria-hidden="true" tabindex="-1"></a>ps <span class="sc">%&gt;%</span></span>
<span id="cb28-3"><a href="downstream-analysis.html#cb28-3" aria-hidden="true" tabindex="-1"></a><span class="fu">transform_sample_counts</span>(., <span class="cf">function</span>(x) x<span class="sc">/</span><span class="fu">sum</span>(x)) <span class="sc">%&gt;%</span></span>
<span id="cb28-4"><a href="downstream-analysis.html#cb28-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_ordination</span>(., <span class="fu">ordinate</span>(., <span class="at">method=</span><span class="st">&quot;PCoA&quot;</span>, <span class="at">distance=</span><span class="st">&quot;bray&quot;</span>),</span>
<span id="cb28-5"><a href="downstream-analysis.html#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="at">color=</span><span class="st">&quot;time&quot;</span> ,<span class="at">title=</span><span class="st">&quot;Bray PCoA (Relative Abundance)&quot;</span>)<span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values=</span>(<span class="fu">c</span>(<span class="st">&quot;tomato&quot;</span>,<span class="st">&quot;blue&quot;</span>))) <span class="sc">+</span> <span class="fu">geom_point</span>()<span class="co">#%&gt;% #+ theme_bw()</span></span></code></pre></div>
<p><img src="16sworkshop_files/figure-html/unnamed-chunk-3-2.png" width="672" /></p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="downstream-analysis.html#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># plot ordination with log transfromed values</span></span>
<span id="cb29-2"><a href="downstream-analysis.html#cb29-2" aria-hidden="true" tabindex="-1"></a>ps <span class="sc">%&gt;%</span></span>
<span id="cb29-3"><a href="downstream-analysis.html#cb29-3" aria-hidden="true" tabindex="-1"></a><span class="fu">transform_sample_counts</span>(., <span class="cf">function</span>(x) <span class="fu">log</span>(<span class="dv">1</span> <span class="sc">+</span> x)) <span class="sc">%&gt;%</span></span>
<span id="cb29-4"><a href="downstream-analysis.html#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_ordination</span>(., <span class="fu">ordinate</span>(., <span class="at">method=</span><span class="st">&quot;PCoA&quot;</span>, <span class="at">distance=</span><span class="st">&quot;bray&quot;</span>),</span>
<span id="cb29-5"><a href="downstream-analysis.html#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="at">color=</span><span class="st">&quot;time&quot;</span> ,<span class="at">title=</span><span class="st">&quot;Bray PCoA (log transfomed)&quot;</span>)<span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values=</span>(<span class="fu">c</span>(<span class="st">&quot;tomato&quot;</span>,<span class="st">&quot;blue&quot;</span>))) <span class="sc">+</span> <span class="fu">geom_point</span>()<span class="co">#%&gt;% #+ theme_bw()</span></span></code></pre></div>
<p><img src="16sworkshop_files/figure-html/unnamed-chunk-3-3.png" width="672" /></p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="downstream-analysis.html#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Permanova ()</span></span>
<span id="cb30-2"><a href="downstream-analysis.html#cb30-2" aria-hidden="true" tabindex="-1"></a>ps_rel<span class="ot">&lt;-</span> ps <span class="sc">%&gt;%</span> <span class="fu">transform_sample_counts</span>(., <span class="cf">function</span>(x) x<span class="sc">/</span><span class="fu">sum</span>(x))</span>
<span id="cb30-3"><a href="downstream-analysis.html#cb30-3" aria-hidden="true" tabindex="-1"></a>asv <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(ps_rel<span class="sc">@</span>otu_table<span class="sc">@</span>.Data)</span>
<span id="cb30-4"><a href="downstream-analysis.html#cb30-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(asv)<span class="ot">&lt;-</span><span class="fu">paste0</span>(ps_rel<span class="sc">@</span>tax_table[,<span class="dv">5</span>],<span class="st">&quot;_&quot;</span>,ps_rel<span class="sc">@</span>tax_table[,<span class="dv">6</span>],<span class="st">&quot;_&quot;</span>,ps_rel<span class="sc">@</span>tax_table[,<span class="dv">7</span>])</span>
<span id="cb30-5"><a href="downstream-analysis.html#cb30-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-6"><a href="downstream-analysis.html#cb30-6" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(ps_rel<span class="sc">@</span>sam_data)</span>
<span id="cb30-7"><a href="downstream-analysis.html#cb30-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-8"><a href="downstream-analysis.html#cb30-8" aria-hidden="true" tabindex="-1"></a>permanova <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">adonis</span>(asv <span class="sc">~</span> time,</span>
<span id="cb30-9"><a href="downstream-analysis.html#cb30-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">data =</span> meta, <span class="at">permutations=</span><span class="dv">999</span>, <span class="at">method =</span> <span class="st">&quot;bray&quot;</span>)</span>
<span id="cb30-10"><a href="downstream-analysis.html#cb30-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-11"><a href="downstream-analysis.html#cb30-11" aria-hidden="true" tabindex="-1"></a>coef <span class="ot">&lt;-</span> <span class="fu">coefficients</span>(permanova)[<span class="st">&quot;time1&quot;</span>,]</span>
<span id="cb30-12"><a href="downstream-analysis.html#cb30-12" aria-hidden="true" tabindex="-1"></a>top.coef <span class="ot">&lt;-</span> coef[<span class="fu">rev</span>(<span class="fu">order</span>(<span class="fu">abs</span>(coef)))[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]]</span>
<span id="cb30-13"><a href="downstream-analysis.html#cb30-13" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">24</span>, <span class="dv">2</span>, <span class="dv">1</span>),<span class="at">cex=</span><span class="fl">0.5</span>)</span>
<span id="cb30-14"><a href="downstream-analysis.html#cb30-14" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(<span class="fu">sort</span>(top.coef), <span class="at">horiz =</span> T, <span class="at">las =</span> <span class="dv">1</span>, <span class="at">main =</span> <span class="st">&quot;Top taxa&quot;</span>)</span></code></pre></div>
<p><img src="16sworkshop_files/figure-html/unnamed-chunk-3-4.png" width="672" /></p>
</div>
<div id="differential-abundance-analysis" class="section level2" number="5.4">
<h2><span class="header-section-number">5.4</span> Differential Abundance Analysis</h2>
<p>Differential abundance analysis is performed between two condition or time points to find taxa that have either bloomed or depleted from one condition compared to another.
There is a variety of tests that can be used to perform this analysis like <strong>DESeq2</strong>, <strong>ANCOM-BC</strong>, <strong>Corncob</strong> to name a few.</p>
<div id="deseq2" class="section level3" number="5.4.1">
<h3><span class="header-section-number">5.4.1</span> DESEQ2</h3>
<p>For demonstration purpose, we will use DESeq2 here.It internally performs data normalization and does p-value correction. It used a negative binomial distribution to detect differences in read counts between groups.To fully understand the method you can watch this detailed video by Statquest <a href="https://www.youtube.com/watch?v=UFB993xufUU&amp;t=12s&amp;ab_channel=StatQuestwithJoshStarmer">here</a></p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="downstream-analysis.html#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span></code></pre></div>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## 
## Attaching package: &#39;BiocGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     combine, intersect, setdiff, union</code></pre>
<pre><code>## The following object is masked from &#39;package:limma&#39;:
## 
##     plotMA</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
##     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
##     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
##     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
##     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which.max, which.min</code></pre>
<pre><code>## 
## Attaching package: &#39;S4Vectors&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     first, rename</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     expand.grid, I, unname</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## 
## Attaching package: &#39;IRanges&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:phyloseq&#39;:
## 
##     distance</code></pre>
<pre><code>## The following objects are masked from &#39;package:dplyr&#39;:
## 
##     collapse, desc, slice</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Loading required package: SummarizedExperiment</code></pre>
<pre><code>## Loading required package: MatrixGenerics</code></pre>
<pre><code>## Loading required package: matrixStats</code></pre>
<pre><code>## 
## Attaching package: &#39;matrixStats&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:dplyr&#39;:
## 
##     count</code></pre>
<pre><code>## 
## Attaching package: &#39;MatrixGenerics&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:matrixStats&#39;:
## 
##     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,
##     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,
##     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,
##     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,
##     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,
##     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,
##     colWeightedMeans, colWeightedMedians, colWeightedSds,
##     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,
##     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,
##     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,
##     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,
##     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,
##     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,
##     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,
##     rowWeightedSds, rowWeightedVars</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view with
##     &#39;browseVignettes()&#39;. To cite Bioconductor, see
##     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</code></pre>
<pre><code>## 
## Attaching package: &#39;Biobase&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:MatrixGenerics&#39;:
## 
##     rowMedians</code></pre>
<pre><code>## The following objects are masked from &#39;package:matrixStats&#39;:
## 
##     anyMissing, rowMedians</code></pre>
<pre><code>## The following object is masked from &#39;package:phyloseq&#39;:
## 
##     sampleNames</code></pre>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="downstream-analysis.html#cb62-1" aria-hidden="true" tabindex="-1"></a>dds<span class="ot">&lt;-</span><span class="fu">phyloseq_to_deseq2</span>(ps,<span class="at">design =</span> <span class="sc">~</span>time)</span></code></pre></div>
<pre><code>## converting counts to integer mode</code></pre>
<pre><code>## Warning in DESeqDataSet(se, design = design, ignoreRank): some variables in
## design formula are characters, converting to factors</code></pre>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="downstream-analysis.html#cb65-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span></code></pre></div>
<pre><code>## estimating size factors</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## gene-wise dispersion estimates</code></pre>
<pre><code>## mean-dispersion relationship</code></pre>
<pre><code>## final dispersion estimates</code></pre>
<pre><code>## fitting model and testing</code></pre>
<pre><code>## -- replacing outliers and refitting for 22 genes
## -- DESeq argument &#39;minReplicatesForReplace&#39; = 7 
## -- original counts are preserved in counts(dds)</code></pre>
<pre><code>## estimating dispersions</code></pre>
<pre><code>## fitting model and testing</code></pre>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="downstream-analysis.html#cb75-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds)</span>
<span id="cb75-2"><a href="downstream-analysis.html#cb75-2" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(res)</span>
<span id="cb75-3"><a href="downstream-analysis.html#cb75-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Orders the rows of data frame in increasing order firstly based on column</span></span>
<span id="cb75-4"><a href="downstream-analysis.html#cb75-4" aria-hidden="true" tabindex="-1"></a><span class="co"># &quot;log2FoldChange&quot; and secondly based on &quot;padj&quot; column</span></span>
<span id="cb75-5"><a href="downstream-analysis.html#cb75-5" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(log2FoldChange, padj)</span></code></pre></div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="dada2.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="project.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["16sworkshop.pdf", "16sworkshop.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
