---
title: "16s rRNA analysis workshop"
author: "Hena R. Ramay"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
description: "This tutorial covers workshop material for all three days"
---

# Introduction

Placeholder


## Workshop Schedule
## Important links

<!--chapter:end:index.Rmd-->


# Basic Information

Placeholder


## What are we trying to achieve
## Basics & Background

<!--chapter:end:02-Introduction.Rmd-->


# Primer Removal

Placeholder


## CutAdapt
### Primer removal for paired end reads
### Quality trimming
### Call cutAdapt from R

<!--chapter:end:03-Primer_removal.Rmd-->


# DADA2


## DADA2 pipeline (v1.2)

From now on, we will be working on the DADA2 package version 1.12. DADA2 has great documentation and an excellent tutorial online.Please go to the following link http://benjjneb.github.io/dada2/tutorial.html

All the notes from now on are my additional comments to help you follow the pipeline:

### Data for the tutorial

The data to use for the tutorial can be downloaded from [here](MiSeqSOPData.zip)

### Getting ready ( load packages and get file list)

Functions that we will be using here are :

- list.files()
- sort()
- strsplit()
- basename()
- sapply()

### Inspect read quality profiles

Read in files

```{r,readdata}
library(dada2);
library(limma)
path <- "MiSeq_SOP/"
list.files(path)
```



```{r, findsamplenames, echo=T}
# Forward and reverse fastq filenames have format: SAMPLENAME_R1_001.fastq and SAMPLENAME_R2_001.fastq
fnFs <- sort(list.files(path, pattern="_R1_001.fastq", full.names = TRUE))
fnRs <- sort(list.files(path, pattern="_R2_001.fastq", full.names = TRUE))
# Extract sample names, assuming filenames have format: SAMPLENAME_XXX.fastq


tmp<-strsplit(basename(fnFs), "_")


# As tmp is a list of vectors, we will have to use sapply function to get the first position out from each vector. The symbol '[' tells sapplly function that it is a vector and 1 specifies the vector index

samples.names<-sapply(tmp,'[',1)

```


### Lets make plots to look at the quality for multiple files


```{r, make_plots,echo=T}

plotQualityProfile(fnFs[1:2])

joint_fnames<-c(rbind(fnFs,fnRs))

plotQualityProfile(joint_fnames)

num=ceiling(length(joint_fnames)/6)

for(i in 1: num)

{
  plotQualityProfile(joint_fnames[i:i+5])


}


```




<!--chapter:end:04-DADA2.Rmd-->


# Day Two

Placeholder


## Main concepts to be discussed:
## Alpha & beta Diversity

<!--chapter:end:05-Phyloseq.Rmd-->

# Summary


<!--chapter:end:06-Project.Rmd-->


# What Next?

<!--chapter:end:07-What_to_learn_next.Rmd-->

